pyiceberg/table/snapshots.py

```diff
@@ -421,8 +421,16 @@ def set_when_positive(properties: Dict[str, str], num: int, property_name: str)
 
 def ancestors_of(current_snapshot: Optional[Snapshot], table_metadata: TableMetadata) -> Iterable[Snapshot]:
     """Get the ancestors of and including the given snapshot."""
-    if current_snapshot:
-        yield current_snapshot
-        if current_snapshot.parent_snapshot_id is not None:
-            if parent := table_metadata.snapshot_by_id(current_snapshot.parent_snapshot_id):
-                yield from ancestors_of(parent, table_metadata)
+    seen = set()
```

Comment:
Out of curiosity, why do we need to track the seen ones? The metadata should be a tree, and not a graph (no cycles)

Reply:
You are right, this was added because I thought there was an infinite loop occurring.